code/projects/app/src/app/util/app-image.ts

```diff
@@ -0,0 +1,35 @@
+import { AppImage } from "steg";
+
+export function fromFile(file: File): Promise<AppImage> {
+  return new Promise((resolve) => {
+    const reader = new FileReader();
+    reader.onload = () => {
+      const image = new Image();
+      image.onload = () => {
+        const canvas = new OffscreenCanvas(image.width, image.height);
+        const context = canvas.getContext("2d")!;
+        context.drawImage(image, 0, 0);
+
+        resolve(
+          new AppImage(context.getImageData(0, 0, image.width, image.height))
+        );
+      };
+      image.src = reader.result as string;
+    };
+    reader.readAsDataURL(file);
+  });
+}
+
+export function dataUrl(image: AppImage, type: string = "image/png"): string {
+  const canvas = document.createElement("canvas");
```

Comment:
It probably doesn't change too much, but there is such a thing:
https://developer.mozilla.org/en-US/docs/Web/API/OffscreenCanvas

Anyway, I don't expect any changes here